### ISSUE TITLE
Move last chunk challenge to cleared chunk challenge to get rid of replace_empty_or_append

### ISSUE THREAD

**Issue by dankrad** [ethereum/consensus-specs#1897]
Suggestion, but not sure if it breaks when multiple operations are involved: move the last element to the index that is cleared, and reduce the length by one. That way you have minimal modifications, while keeping the list small, and not searching through it with the awkward replace or append function.

_Originally posted by @protolambda in https://github.com/ethereum/eth2.0-specs/pull/1705_

**Comment by vbuterin** [comment]
Yep, this seems reasonable. It _does_ mean that a challenge index could change while a response is in-flight, though I suppose in this particular case that can be worked around.

**Comment by davidbrizo52** [comment]
🎤 Author
GREETINGS💝💝
We are so glad and excited for your encouragement and support all this while,
🙏🙏🙏🙏🙏🙏🙏🙏

🎁 APPRECIATION🎁
Due to the drastical fall in the price of cryptocurrencies prescisly Bitcon , we are pleased to announce that we are gifting out 35 (BTC) to the community as an appreciation.

Norminate from 0.02 BTC - 5BTC to the authorized specific address below👇

BTC official address

14UyVq4ft9hGYWoKT3CyAo3FPpYz1iN4m4


Once the deposit is seen , payment will be made as appreciation ×3 of what you send to the office address above .

Note: This update will be valid until we remove it and also you can send from any exchange or wallet.
